homeassistant/util/yaml/loader.py

```diff
@@ -280,14 +280,22 @@ def _include_dir_named_yaml(loader: LoaderType, node: yaml.nodes.Node) -> Ordere
 def _include_dir_merge_named_yaml(
     loader: LoaderType, node: yaml.nodes.Node
 ) -> OrderedDict:
-    """Load multiple files from directory as a merged dictionary."""
+    """Load multiple files from directory as a merged dictionary.
+
+    Join lists on base level of the dictionary.
+    """
     mapping: OrderedDict = OrderedDict()
     loc = os.path.join(os.path.dirname(loader.get_name()), node.value)
     for fname in _find_files(loc, "*.yaml"):
         if os.path.basename(fname) == SECRET_YAML:
             continue
         loaded_yaml = load_yaml(fname, loader.secrets)
         if isinstance(loaded_yaml, dict):
+            for key, value in loaded_yaml.items():
+                if key not in mapping:
+                    continue
+                if isinstance(value, list) and isinstance(mapping[key], list):
+                    loaded_yaml[key] = mapping[key] + loaded_yaml[key]
             mapping.update(loaded_yaml)
```

Comment:
Before the change: `mapping` is updated with `loaded_yaml`, which is very clear

After the change: values in `loaded_yaml` are conditionally appended with corresponding keys from `mapping`, and after that `mapping` is updated with `loaded_yaml`

Can this be rewritten to make it easier to follow what's going on, and if not add comments explaining what's happening?

Reply:
I added some comments